extensions/amp-story/0.1/progress-bar.js

```diff
@@ -112,45 +118,59 @@ export class ProgressBar {
    * @private
    */
   assertValidPageIndex_(pageIndex) {
-    dev().assert(pageIndex >= 0 && pageIndex < this.pageCount_,
-        `Page index ${pageIndex} is not between 0 and ${this.pageCount_}.`);
+    dev().assert(pageIndex >= 0 && pageIndex < this.segmentCount_,
+        `Page index ${pageIndex} is not between 0 and ${this.segmentCount_}.`);
   }
 
 
   /**
-   * @param {number} pageIndex The index of the new active page.
+   * @param {string} pageId The index of the new active page.
    * @public
    */
-  setActivePageIndex(pageIndex) {
-    this.assertValidPageIndex_(pageIndex);
-    for (let i = 0; i < this.pageCount_; i++) {
-      if (i < pageIndex) {
-        this.updateProgress(i, 1.0, /* withTransition */ i == pageIndex - 1);
+  setActivePageIndex(pageId) {
+    const progressBarIndex = this.segmentIdMap_[pageId];
+
+    for (let i = 0; i < this.segmentCount_; i++) {
+      if (i < progressBarIndex) {
+        this.updateProgressByIndex_(i, 1.0,
+            /* withTransition */ i == progressBarIndex - 1);
       } else {
         // The active page manages its own progress by firing PAGE_PROGRESS
         // events to amp-story.
-        this.updateProgress(i, 0.0, /* withTransition */ (
-          pageIndex != 0 && this.activePageIndex_ != 1));
+        this.updateProgressByIndex_(i, 0.0, /* withTransition */ (
+          progressBarIndex != 0 && this.activePageIndex_ != 1));
       }
     }
   }
 
+
+  /**
+   * The
+   * @param {string} pageId the id of the page whos progress to change
+   * @param {number} progress A number from 0.0 to 1.0, representing the
+   *     progress of the current page.
+   */
+  updateProgress(pageId, progress) {
+    const progressBarIndex = this.segmentIdMap_[pageId];
+    this.updateProgressByIndex_(progressBarIndex, progress);
+  }
+
+
   /**
-   * @param {number} pageIndex The index of the page whose progress should be
+   * @param {number} progressBarIndex The index of the progress bar segment whose progress should be
    *     changed.
    * @param {number} progress A number from 0.0 to 1.0, representing the
    *     progress of the current page.
    * @param {boolean=} withTransition
    * @public
    */
-  updateProgress(pageIndex, progress, withTransition = true) {
-    this.assertValidPageIndex_(pageIndex);
-
-    this.activePageIndex_ = pageIndex;
+  updateProgressByIndex_(progressBarIndex, progress, withTransition = true) {
+    this.assertValidPageIndex_(progressBarIndex);
```

Comment:
i don't think the assertion is needed. given the progressBarIndex is locally computed

Reply:
I agree, but I think it's more like we should assert it above when you get it from `this.segmentIdMap_`.  I would actually expect that to be a type error without doing so.

Reply:
@newmuis could you elaborate here? I take this to mean assert `progressBarIndex !== undefined` on line:155. Is that correct?

Reply:
changed assertion to make sure segmentId key exists